src/compiler/types.ts

```diff
@@ -9850,6 +9850,12 @@ export const commentPragmas = {
     },
 } as const;
 
+export const enum JSDocParsingMode {
+    KeepAll = 0,
+    KeepSemanticOnly = 1,
+    SkipAll = 2,

```

Comment:
I feel like "Keep" may be a bad prefix here; "Keep" makes me feel like we're always parsing then _dropping_ from the AST, not always skipping. Perhaps it'd be better to have:

- `All`
- `SkipNonSemantic`
- `SkipAll`

? Anyone else want to paint the bikeshed?

Reply:
I like your new proposal, or

```
JSDoc.ParseAll
JSDoc.SkipNonSemantic
JSDoc.SkipAll
```

(assuming that JSDoc isn't already taken by another value -- I guess `JSDocMode` or `JSDocParsing` are OK too)

Reply:
JSDoc is taken by the namespace. I was also trying to avoid it being "too short" because it _technically_ doesn't control JSDoc parsing under the hood at all. It's more like... `JSDocSkipMode`?

Reply:
`Semantic` is also a little fuzzy. Maybe `SkipNonChecked` or `ParseCheckedTagsOnly`

Reply:
"Semantic" is the name we spitballed a day or two ago, and what I referred to it as before this PR. I am of course happy to come to a consensus on any new name because we'll be stuck with it forever!

Reply:
I've renamed things further; check the PR description. They're now close to your suggestion with the nuance about errors/no errors.